### PR TITLE
Regarding Issue 68

### PR DESCRIPTION
**Pull request by Knlsharma**
**Close** #68 . Here I've added **DFS** and **TernarySearchTree**

### CODE REVIEW

**Comment by rishabhgarg25699** [comment]
@Knlsharma Do proper indentation and remove extra spaces

**Comment by Knlsharma** [comment]
> @Knlsharma Do proper indentation and remove extra spaces

Okay

**Comment by Knlsharma** [comment]
> @Knlsharma Do proper indentation and remove extra spaces

Hey, @rishabhgarg25699  Please Checkout the above changes!

File: Searching Algo/TernarySearchTree.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	{
+        if (x == null)
+            return null;
+
+        if (key.length() == 0)
+             throw new IllegalArgumentException("key must have length >= 1");
+
+        char c = key.charAt(d);
+        if (c < x.c)            
+		{
```
Comment (rishabhgarg25699) [inline Searching Algo/TernarySearchTree.java]: No proper indentation

File: Searching Algo/TernarySearchTree.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        else if (c > x.c)   
+		{		
+	 return get(x.right, key, d);
+		}
+        else if (d < key.length() - 1) 
+		{ 
+	   return get(x.mid,   key, d+1);
+		}
+        else            
+		{
```
Comment (rishabhgarg25699) [inline Searching Algo/TernarySearchTree.java]: Indentation

File: Searching Algo/TernarySearchTree.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            size++;
+
+        root = put(root, key, val, 0);
+    }
+
+    private Node<Value> put(Node<Value> x, String key, Value val, int d) 
+	{
+        char c = key.charAt(d);
+        if (x == null) 
+		{
```
Comment (rishabhgarg25699) [inline Searching Algo/TernarySearchTree.java]: Indentation Problem

**Comment by rishabhgarg25699** [comment]
Many indentation problem

**Comment by Knlsharma** [comment]
@rishabhgarg25699   indented towards left.

Hope this helps !

File: Searching Algo/Dfs.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+While Space complexity for DFS is O(h) where h is the maximum height of the tree.
+**/
+
+import java.util.*;
+public class DFS
+{
+int V;    
+LinkedList<Integer> adjList[]; 
+DFS(int v)
+{
```
Comment (rishabhgarg25699) [inline Searching Algo/Dfs.java]: @Knlsharma Where is the indentation. All program is in the single line indentation.

**Comment by rishabhgarg25699** [comment]
@Knlsharma no 4 space indentation.

**Comment by Knlsharma** [comment]
Hey, this time I have also beautified so that indentation and readability enhance.

**I mean if I am getting understood correctly.**

**Review by rishabhgarg25699 — APPROVED** [review]

**Pull request merged**